core/commands/add.go

```diff
@@ -227,9 +250,24 @@ func add(n *core.IpfsNode, reader io.Reader) (*dag.Node, error) {
 	return node, nil
 }
 
-func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool) (*dag.Node, error) {
+func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool, hidden bool, ignoreFilePatterns []ignore.GitIgnore) (*dag.Node, error) {
```

Comment:
@jbenet agreed, when I added two extra inputs, my brain said "time for struct". But I was hesitant to make/break any other existing patterns here.

It seems like most of the CLI methods are operating on _"list of Parameters"_ rather than _"struct that wraps Parameters"_. Honestly, that may be a good fix in general.